framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -254,6 +255,46 @@ public void insertValue(FlowExpressions.Receiver r, AnnotationMirror a) {
         insertValue(r, analysis.createSingleAnnotationValue(a, r.getType()));
     }
 
+    /**
+     * Add the annotation {@code a} for the expression {@code r} (correctly deciding where to store
+     * the information depending on the type of the expression {@code r}).
+     *
+     * <p>This method does not take care of removing other information that might be influenced by
+     * changes to certain parts of the state.
+     *
+     * <p>If there is already a value {@code v} present for {@code r}, then the greatest lower bound
+     * of the new and old value is inserted into the store unless it's bottom. If the greates lower
+     * bound is bottom, then the new value is inserted.
```

Comment:
What is the intuition behind this special treatment for the bottom qualifier? I don't see why this is different from other GLBs.

Reply:
For qualifiers that take arguments that don't override `QualifierHierarchy#greatestLowerBound`,  glb is not actually the *greatest* lower bound, but rather a lower bound.  That's why it's not safe to add bottom.  (I'll update the Javadoc.)

Reply:
If bottom, it might be from an incorrect answer and to be conservative about the incorrect value we forbid bottom.